belgian layout had no sendstring definition

kimat opened the pull request:
Note that you have to send the dead keys twice: tilda, circ.

Tested with the following : 

```c
SEND_STRING(" !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^^``_abcdefghijklmnopqrstuvwxyz{|}~~");
```

And what the keyboard typed when triggered:

```
 !"#$%&'()*+,-.+0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_abcdefghijklmnopqrstuvwxyz{|}
```

Note that the deadkey backtick (grave accent) isn't displayed for some reason

noroadsleft commented:
@kimat Welcome!

Am I correct in understanding that these changes don't fully enable the full sendstring charset, because some of the characters are used as dead keys in Belgium? I'm American so my experience here is limited. As I recall, to get the character of a dead key, rather than combine it with a key to get an accented character, you need to use the dead key followed by Space. Is that correct? Trying to think of a solution for the future.

fauxpark commented on quantum/keymap_extras/sendstring_belgian.h:
```diff
@@ -0,0 +1,94 @@
+    1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 0, 0, 0, 0, 1,
+    1, 0, 0, 0, 0, 0, 0, 0,
```
```suggestion
    0, 0, 0, 0, 0, 0, 0, 0,
```

fauxpark commented on quantum/keymap_extras/sendstring_belgian.h:
```diff
@@ -0,0 +1,94 @@
+    0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 1, 1, 1, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0,
```
```suggestion
    1, 0, 0, 0, 0, 0, 0, 0,
```

fauxpark commented on quantum/keymap_extras/sendstring_belgian.h:
```diff
@@ -0,0 +1,94 @@
+ /* SPACE  !     "      #        $     %     &      '                 */
+    KC_SPC, BE_EXLM, BE_QUOT, BE_QUOT, BE_DLR,BE_UGRV,BE_AMP, BE_APOS ,
+  /*  (      )     *      +        ,        -        .       /        */
+    BE_LPRN, BE_RPRN, BE_DLR,BE_EQL, BE_COMM, BE_MINS, BE_SCLN, BE_EQL,
```
```suggestion
    BE_LPRN, BE_RPRN, BE_DLR,BE_EQL, BE_COMM, BE_MINS, BE_SCLN, BE_COLN,
```

fauxpark commented on quantum/keymap_extras/sendstring_belgian.h:
```diff
@@ -0,0 +1,94 @@
+  /*  (      )     *      +        ,        -        .       /        */
+    BE_LPRN, BE_RPRN, BE_DLR,BE_EQL, BE_COMM, BE_MINS, BE_SCLN, BE_EQL,
+  /*   0     1     2     3     4     5    6     7                     */
+    KC_0, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7,
```
```suggestion
    BE_AGRV, BE_AMP, BE_EACU, BE_QUOT, BE_APOS, BE_LPRN, BE_PARA, BE_EGRV,
```

fauxpark commented on quantum/keymap_extras/sendstring_belgian.h:
```diff
@@ -0,0 +1,94 @@
+  /*   0     1     2     3     4     5    6     7                     */
+    KC_0, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7,
+  /*   8     9      :       ;         <        =       >       ?      */
+    KC_8, KC_9, BE_COLN, BE_SCLN, BE_LESS,    BE_EQL, BE_LESS, BE_COMM,
```
```suggestion
    BE_EXLM, BE_CCED, BE_COLN, BE_SCLN, BE_LESS,    BE_EQL, BE_LESS, BE_COMM,
```

fauxpark commented on quantum/keymap_extras/sendstring_belgian.h:
```diff
@@ -0,0 +1,94 @@
+  /*   8     9      :       ;         <        =       >       ?      */
+    KC_8, KC_9, BE_COLN, BE_SCLN, BE_LESS,    BE_EQL, BE_LESS, BE_COMM,
+  /*   @     A     B     C     D     E     F     G                    */
+    KC_2, BE_A, KC_B, KC_C, KC_D, KC_E, KC_F, KC_G,
```
```suggestion
    BE_EACU, BE_A, KC_B, KC_C, KC_D, KC_E, KC_F, KC_G,
```

fauxpark commented on quantum/keymap_extras/sendstring_belgian.h:
```diff
@@ -0,0 +1,94 @@
+  /*   X     Y     Z      [         \       ]       ^      _          */
+    KC_X, KC_Y, BE_Z, BE_CIRC, BE_LESS, BE_DLR, BE_CIRC, BE_MINS,
+  /*   `       a     b     c     d     e     f     g                  */
+    0, BE_A, KC_B, KC_C, KC_D, KC_E, KC_F, KC_G,
```
```suggestion
    BE_MU, BE_A, KC_B, KC_C, KC_D, KC_E, KC_F, KC_G,
```

fauxpark commented on quantum/keymap_extras/sendstring_belgian.h:
```diff
@@ -0,0 +1,94 @@
+  /*    p    q     r     s     t     u     v     w                    */
+    KC_P, BE_Q, KC_R, KC_S, KC_T, KC_U, KC_V, BE_W,
+  /*    x    y     z      {        |        }        ~    DELETE      */
+    KC_X, KC_Y, BE_Z, KC_9, KC_1, KC_0, BE_EQL, KC_DEL
```
```suggestion
    KC_X, KC_Y, BE_Z, BE_CCED, BE_AMP, BE_AGRV, BE_EQL, KC_DEL
```

fauxpark commented:
The Belgian keyboard layout (red are dead keys):

![](https://upload.wikimedia.org/wikipedia/commons/9/93/Belgian_keyboard_layout.png)

kimat commented:
I fixed backtick which I forgot to define... And I fixed slash. Thanks @fauxpark. 

@fauxpark I for one think of @ as altgr+2 and not altgr+é, and I'd argue this makes makes the current format easier to read.

@noroadsleft thanks, 
The full sendstring charset can be sent now. It does require doubling, which is what you would naturally do when typing, or adding a space after dead keys to have them display. Something better could be envisionned but this works already.
Sendstring could actually support more keys than the 128 ascii chars by combining e.g. ralt and shift or allowing e.g. BE_EACU through without modifiers.

fauxpark commented:
I see what you mean, but in this layout `2` is Shift+`é`, so AltGr+`2` would actually mean AltGr+Shift+`é`. It makes much more sense to me to use the base unshifted language-specific keycodes here to demonstrate why the Shift and AltGr tables exist.

kimat commented:
So we are down to two dead keys ~ and ` that need an extra space in the sendstring but we have the full sendstring charset.

drashna commented:
Just to make sure, is this read to go (as is), or does it need more time?

kimat commented:
@drashna It's ready to go.

drashna commented:
Awesome, thanks!
